Doc/reference/expressions.rst

```diff
@@ -83,18 +83,137 @@ exception.
-or more underscores, it is considered a :dfn:`private name` of that class.
-Private names are transformed to a longer form before code is generated for
-them.  The transformation inserts the class name, with leading underscores
-removed and a single underscore inserted, in front of the name.  For example,
-the identifier ``__spam`` occurring in a class named ``Ham`` will be transformed
-to ``_Ham__spam``.  This transformation is independent of the syntactical
-context in which the identifier is used.  If the transformed name is extremely
-long (longer than 255 characters), implementation defined truncation may happen.
-If the class name consists only of underscores, no transformation is done.
+Private name mangling
+^^^^^^^^^^^^^^^^^^^^^
 
+When an identifier that textually occurs in a class definition begins with two
+or more underscore characters and does not end in two or more underscores, it
+is considered a :dfn:`private name` of that class.
+
+.. seealso::
+
+   The :ref:`tutorial on classes <tut-classdefinition>` for more details
+   on classes and their declaration.
+
+More precisely, private names are transformed to a longer form before code is
+generated for them.  If the transformed name is longer than 255 characters,
+implementation-defined truncation may happen.
+
+The transformation is independent of the syntactical context in which the
+identifier is used and the transformation rule is defined as follows:
+
+- If the class name consists only of underscores, the transformation is the
+  identity, e.g., the identifier ``__spam`` occurring in a class named ``_``
+  or ``__`` is left as is.
+
+- Otherwise, the transformation inserts the class name, with leading
+  underscores removed and a single underscore inserted, in front of
+  the identifier, e.g., the identifier ``__spam`` occurring in a class
+  named ``Foo``, ``_Foo`` or ``__Foo`` is transformed to ``_Foo__spam``.
+
+.. _private-name-mangling-access:
+
+.. rubric:: Accessing members with mangled names
+
+The corresponding private member is accessed by its defining class using
+its non-transformed name. On the other hand, the transformed name must be
+used to access the private member *externally* (e.g., in a function or in
+a subclass):
+
+.. code-block:: python
+
+   class A:
+       def __one(self):
+           return 1
+       def two(self):
+           return 2 * self.__one()
+
+   class B(A):
+       def three(self):
+           return 3 * self._A__one()
+
+   four = 4 * A()._A__one()
+
+.. seealso::
+
+   The tutorial on :ref:`private variables <tut-private>` in classes for
+   a more complete example.
+
+.. _private-name-mangling-imports:
+
+.. rubric:: Mangled names in imports
+
+For identifiers declared using :keyword:`import` statements, the mangling
+rule is slightly different. Throughout this paragraph, assume that we have
+the following filesystem layout and, unless stated otherwise, the code snippets
+are in the ``__main__.py`` file.
```

Comment:
No need to complicate this sentence when all snippets are in `__main__.py`.

```suggestion
the following filesystem layout and the code snippets are in the
``__main__.py`` file.
```

Reply:
Ah yes, I wasn't sure whether I'll need a special case but forgot to update the text.